src/lifecycle.js

```diff
@@ -366,27 +383,67 @@ function triggerLifecycle (target, component) {
  * @returns {undefined}
  */
 function initElements (elements) {
-  var elementsLen = elements.length;
-
-  for (var a = 0; a < elementsLen; a++) {
-    var element = elements[a];
-
+  var a, b, c, d, e;
```

Comment:
I'd probably just reuse `i` and `j` across the different for loops.

I think declaring all these as the first line is a little jarring if you haven't seen the rest of the function before. So I'd also say declare & init them in the `for` statements.